VocabularyTrainer/Home Screen/HomeViewController.swift

```diff
@@ -0,0 +1,249 @@
+    }
+
+    required init?(coder: NSCoder) { nil }
+
+    // MARK: - View Lifecycle
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        removeNavigationBarBackground()
+    }
+
+    // MARK: - Private Methods
+
+    private func setupView() {
+        view.backgroundColor = HomeViewModel.Colors.background
+        view.addSubview(collectionView)
+        view.addSubview(headerView)
+
+        collectionView.setCollectionViewLayout(viewModel.layout, animated: true)
+
+        NSLayoutConstraint.activate([
+            headerView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: horizontalCollectionViewMargins),
+            headerView.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor, constant: Layout.defaultMargin),
+            headerView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -horizontalCollectionViewMargins),
+
+            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: horizontalCollectionViewMargins),
+            collectionView.topAnchor.constraint(equalTo: headerView.bottomAnchor, constant: 24),
+            collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -horizontalCollectionViewMargins),
+            collectionView.bottomAnchor.constraint(equalTo: view.layoutMarginsGuide.bottomAnchor)
+        ])
+        applyCollectionViewChanges()
+    }
+
+    /// Applies changes of data source.
+    private func applyCollectionViewChanges() {
+        var snap = datasource.snapshot()
+        if !snap.sectionIdentifiers.isEmpty {
+            snap.deleteSections([0])
+        }
+        snap.appendSections([0])
+        snap.appendItems(viewModel.data)
+        datasource.apply(snap)
+    }
+
+    /// Sets `navigationItem` with Flippy logo and import/export buttons.
+    private func setNavigationItem() {
+        navigationItem.leftBarButtonItem = navbarLogo
+        let importButton = UIBarButtonItem(customView: iconButton(
+            text: Strings.importButtonTitle,
+            trailingImage: UIImage(systemName: "square.and.arrow.down"),
+            tapHandler: { [weak self] in
+                self?.tappedImport()
+            }
+        ))
+        let exportButton = UIBarButtonItem(customView: iconButton(
+            text: Strings.exportButtonTitle,
+            trailingImage: UIImage(systemName: "square.and.arrow.up"),
+            tapHandler: { [weak self] in
+                self?.tappedExport()
+            }
+        ))
+        navigationItem.rightBarButtonItems = [exportButton, importButton]
+    }
+
+    /// Creates and returns button with trailing image.
+    /// - Parameters:
+    ///   - text: The text to show.
+    ///   - image: The trailing image shown next to the text.
+    ///   - tapHandler: The block, executed when tapping the button.
+    private func iconButton(text: String, trailingImage image: UIImage?, tapHandler: (() -> Void)?) -> UIButton {
```

Comment:
Maybe we could add it in a UIButton extension?

Reply:
Good idea thought about that too, but ditched the idea as I thought it's a special thing for that VC.

Anyway, I'll add it :) makes sense to keep the VCs simple and short.

Maybe its about time to start that separate UI-package for SPM :D

Reply:
added extension in: ce89cce01efffc572f02ee6f977e84b1f3c25bcb